### DISCUSSION TITLE
Bug List

### DISCUSSION THREAD

**Discussion by alvarix** [alvarix/FLogger#129] [General]
@chadcrume let's gather bugs here. 

- [x] Pretext Error
Attaching a screenshot of flogger and the file.
The pretext = Workout 2.1
<img width="1536" alt="Screenshot 2024-12-21 at 11 27 07 AM" src="https://github.com/user-attachments/assets/f944f55d-648d-4a95-a3f1-af81fd643f46" />
[Workout-2.1.flogger.txt](https://github.com/user-attachments/files/18219247/Workout-2.1.flogger.txt)

The errors are the same ones I noted on our call that appear on all pages
useDropboxFiles.ts:36 
       POST https://api.dropboxapi.com/oauth2/token?grant_type=authorization_code&code=3ndr0N59MEQAAAAAAAJVpVmYZncdf6saBpIUfu6ZLW0&client_id=85vbmd9vlyyb5kp&code_verifier=MTQwLDcsOTAsOTIsODEsNzcsNTUsMjE4LDIzMywzNCw1MSwxNTMsMTIxLDIyOSwyNSw4OSwyMDEsMTczLDEyMyw4Myw5OCwyNiwyNDMsMjU1LDE0MCwxNDYsMTcxLDg1&redirect_uri=http://localhost:5173/dbauthpopup/ 400 (Bad Request)
       
fetch @ useDropboxFiles.ts:36
getAccessTokenFromCode @ dropbox.js?v=b8012b12:1434
useDropboxFiles @ useDropboxFiles.ts:78
(anonymous) @ useDropboxFlogs.ts:38Understand this errorAI
useDropboxFiles.ts:96 Uncaught (in promise) TypeError: Cannot read properties of null (reading 'reload')
    at useDropboxFiles.ts:96:31

**Comment by alvarix** [comment]
## mobile auth frozen

@chadcrume
cant seem to login on any vercel URL

**Comment by alvarix** [comment]
This discussion should be a repo note
@chadcrume

**Comment by chadcrume** [comment]
Light/dark mode reverts to browser setting on reload, with theme button text not matching actual current mode

**Steps to reproduce:**
1. Set browser mode to Dark
2. Load Flogger, set mode to Light
3. Connect to Dropbox
4. After auth pop-up closes, it reloads the Flogger window. The theme is back to Dark mode, while the button says Light.

**Reply by alvarix** [reply]
@chadcrume 
I cant reproduce this in chrome. 
I do notice that it doesnt respect the browser choice (if i switch to dark in settings flogger stays light), 
but it does respect the OS choice (if i switch to dark in settings flogger turns dark).

**Reply by alvarix** [reply]
@chadcrume been digging into this more. 
So I spent a while trying to figure out Chrome dark mode vs system.
**I thing chrome "dark mode" is just the browser UI.**

> Browser UI Dark Mode
The dark mode for the Chrome browser UI affects elements like the toolbar, settings, and homepage. This mode changes the overall appearance of the browser to a darker theme, which is easier on the eyes in low-light conditions. Users can enable this dark theme through Chrome settings or by following their operating system's theme settings[5](https://support.google.com/chrome/answer/9275525?co=GENIE.Platform%3DDesktop&oco=2)[6](https://support.google.com/chrome/answer/9275525?hl=en&co=GENIE.Platform%3DAndroid).
Forced Dark Mode for Websites
In contrast, Forced Dark Mode is a feature introduced in Chrome 78 that allows users to apply a dark theme to any website, regardless of whether that site natively supports dark mode. This is done by inverting colors using color theory principles, resulting in light text on a dark background[1](https://www.zdnet.com/article/how-to-forcibly-enable-dark-mode-in-on-every-website-with-google-chrome/)[2](https://www.wisestamp.com/blog/dark-mode-chrome/). To enable this feature, users must access Chrome flags by entering chrome://flags/#enable-force-dark in the address bar and selecting "Enabled" from the dropdown menu[1](https://www.zdnet.com/article/how-to-forcibly-enable-dark-mode-in-on-every-website-with-google-chrome/)[4](https://superuser.com/questions/1588110/dark-mode-in-chrome-based-browsers).

https://www.perplexity.ai/search/chrome-dark-mode-is-just-brows-k4QSGd3HTFq8dySY.pXCtg


I also tried stripping styles down and couldnt get Chrome dark mode to work. 
Also notice that all the website's here listed as 'nice dark mode' dont change with google's dark mode pref:
https://www.reddit.com/r/webdev/comments/1bqv4b0/your_favourite_dark_mode_website/

There's apparently a flag to turn on true dark mode:
chromium --enable-features=WebContentsForceDark
https://superuser.com/questions/1588110/dark-mode-in-chrome-based-browsers

I'm going to mark this as quote "resolved" for now if you agree.

**Comment by chadcrume** [comment]
Here's what I see:
https://www.loom.com/share/696b4cfc15784e67b2358a1dec7857f5?sid=1c903d63-92bd-4970-a2fd-5cdf3157a1dc

**Reply by alvarix** [reply]
whats this **dark reader** extension?
Does it still misbehave if you disable it?

**Reply by chadcrume** [reply]
In the loom I disabled it just to show that it's not part of the issue

**Comment by alvarix** [comment]
@chadcrume 
Mobile popups blocked. 
I mentioned this elsewhere. 

I've tested in Safari and Chrome. 
In one it silently fails, in the other it notifies that a popup was blocked - but there is no interaction that can make it work I think.
[Screen recording. ](https://www.dropbox.com/scl/fi/cd668d2p0qs0wro9jciuy/flogger-dbx-bug.MP4?rlkey=icykf0wepcnbrk2t0hu54finl&dl=0)

Either we have to find an alternate to a popup or provide instructions (that arent too tedious) to navigate the popup.

It also currently fails on main vercel url - flogger.vercel.app
`Invalid redirect_uri: "https://flogger-h0poo9jxb-alvarixs-projects.vercel.app/dbauthpopup/": It must exactly match one of the redirect URIs you've pre-configured for your app (including the path).`

**Comment by alvarix** [comment]
@chadcrume  
## Parsing Errors persist
[Workout-2.1.flogger.txt](https://github.com/user-attachments/files/18275519/Workout-2.1.flogger.txt)
![Screenshot 2024-12-30 at 9 50 40 AM](https://github.com/user-attachments/assets/d43d99ab-4738-40da-8555-6ec1939d556f)

**Reply by chadcrume** [reply]
The specific issue with the Workout file was fixed. Other work remains:

- [ ] Fix date followed by date
- [ ] Fix saving pretext
- [ ] Make pretext editable

**Comment by chadcrume** [comment]
Public production URL (https://flogger.vercel.app/) sends branch URL to Dropbox:

> Invalid redirect_uri: "https://flogger-h0poo9jxb-alvarixs-projects.vercel.app/dbauthpopup/": It must exactly match one of the redirect URIs you've pre-configured for your app (including the path).

**Comment by chadcrume** [comment]
@alvarix , `Invalid redirect_uri` error in production is fixed with #138

**Reply by alvarix** [reply]
@chadcrume 
cool. 
Any ideas what to do about popups on phone? 
**Safari**
Interestingly when I save it onto my phone's home screen it's not loading any browser UI, it's it's own window with it's own icon separate from teh safari window - acting more like an app. 
I see teh DBx in progress message but no popup or anyway to allow it. 
The link to **popup window** has no effect when clicked.

I wanted to check my ios simulator but it currently wont boot.

**Chrome**
Works on chrome!
It does some stuff fast then loads db files.
Here it behaves normally like a tab in the browser's window.
Good for now but we really want safari to work

**Reply by chadcrume** [reply]
@alvarix 
I don't have an iphone or ipad so troubleshooting is limited. I can emulate iPhone Safari using my MacOS Safari.

<img width="912" alt="Screenshot 2025-01-07 at 6 26 05 PM" src="https://github.com/user-attachments/assets/fbd5c606-e55d-40de-9272-8fda7bc13d48" />

That works for me. I do have to click this icon in the address bar to allow pop-ups when first blocked.
<img width="672" alt="Screenshot 2025-01-07 at 6 43 00 PM" src="https://github.com/user-attachments/assets/6bb355c2-ac96-4398-b94a-1ba730c77f65" />

**Reply by alvarix** [reply]
@chadcrume 
Yeah the in safari emulalor is kinda shit, but the **Simulator** app works better.
It's part of XCode.
You can use the dev tools in Safari desktop to inspect on the Simulator.
<img width="426" alt="Screenshot 2025-01-08 at 9 43 11 AM" src="https://github.com/user-attachments/assets/056658de-8f20-402e-8dcb-4270367a4bbc" />

**Reply by chadcrume** [reply]
This post describes how to get the OS Versions for simulators downloaded for use in Simulator.

https://forums.developer.apple.com/forums/thread/738573

**Comment by alvarix** [comment]
- [x] Buttons overlapping logo on mobile.
@alvarix 
![IMG_136300771DD3-1](https://github.com/user-attachments/assets/bd786080-b481-4970-8781-536979042b23)

**Comment by chadcrume** [comment]
- [ ] Handle dbx session expiration
@chadcrume 
<img width="868" alt="Screenshot 2025-01-07 at 6 11 53 PM" src="https://github.com/user-attachments/assets/670941c3-2fce-407e-9d1c-4defe9a6e2be" />
